PINCache/PINOperationQueue.m

```diff
@@ -0,0 +1,217 @@
+//
+//  PINOperationQueue.m
+//  Pods
+//
+//  Created by Garrett Moon on 8/23/16.
+//
+//
+
+#import "PINOperationQueue.h"
+#import <pthread.h>
+
+@interface NSNumber (PINOperationQueue) <PINOperationReference>
+
+@end
+
+@interface PINOperationQueue () {
+  pthread_mutex_t _lock;
+  //increments with every operation to allow cancelation
+  NSUInteger _operationReferenceCount;
+  
+  dispatch_queue_t _serialQueue;
+  BOOL _serialQueueBusy;
+  
+  dispatch_semaphore_t _concurrentSemaphore;
+  dispatch_queue_t _concurrentQueue;
```

Comment:
@appleguy Honestly, I'm not 100% sure of all the tradeoffs around using the global queue vs a new one. One issue I could see with using the global queue is starvation. I guess I could make it an init option.